grpc_opa/acct_entitlements.go

```diff
@@ -0,0 +1,161 @@
+package grpc_opa_middleware
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus/ctxlogrus"
+	logrus "github.com/sirupsen/logrus"
+
+	"github.com/infobloxopen/atlas-authz-middleware/pkg/json_utils"
+)
+
+const (
+	// DefaultAcctEntitlementsApiPath is default OPA path to fetch acct entitlements
+	DefaultAcctEntitlementsApiPath = "v1/data/authz/rbac/acct_entitlements_api"
+)
+
+// AcctEntitlementsType is a convenience data type, returned by GetAcctEntitlements()
+// (map of acct_id to map of service to array of features)
+type AcctEntitlementsType map[string]map[string][]string
+
+// AcctEntitlementsApiResult is the data type returned by OPA RESTAPI query to acct_entitlements_api
+type AcctEntitlementsApiResult struct {
+	Loggr  *logrus.Logger        `json:"-"`
+	Result *AcctEntitlementsType `json:"result"`
+}
+
+// UnmarshalJSON implements json.Unmarshaler interface for AcctEntitlementsApiResult
+func (u *AcctEntitlementsApiResult) UnmarshalJSON(rawBytes []byte) error {
```

Comment:
This is too fragile. We can get more readable code using the following struct: 

```go
var results = []string{
`{ "result": {
	"acct1": { "svc1a": [ "feat1a1", "feat1a2" ] },
	"acct2": { "svc2a": [ "feat2a1", "feat2a2" ],
		       "svc2b": [ "feat2b1", "feat2b2" ] }
}}`,
// ---
`{ "result": null }`,
// ---
`{ "result": {
	"acct1": { "svc1a": [ "feat1a1", "feat1a2" ] },
	"acct2": null
}}`,
}

type result struct {
	Result map[string]map[string]map[string]interface{}
}

func main() {
	for i, res := range results {
		var r result
		err := json.Unmarshal([]byte(res), &r.Result)
		printj(i, r, err)
	}
}
```

Please play it [here](https://go.dev/play/p/fw1SUwS6mkU).

Reply:
yes I found out custom unmarshaler is actually not required, i've removed it.